internal/cache/file/cache_handler_test.go

```diff
@@ -602,166 +683,293 @@ func Test_InvalidateCache_WhenEntryNotInCache(t *testing.T) {
+				minObj := createObject(t, chTestArgs.bucket, objName, []byte(objContent))
+
+				cacheHandle, err := chTestArgs.cacheHandler.GetCacheHandle(minObj, chTestArgs.bucket, false, 0)
+
+				assert.NoError(t, err)
+				assert.Nil(t, cacheHandle.validateCacheHandle())
+			}
+
+			// Start concurrent GetCacheHandle()
+			for i := 0; i < 5; i++ {
+				wg.Add(1)
+				go invalidateCacheTestFun(i)
+				wg.Add(1)
+				go getCacheHandleTestFun(i)
+			}
+			wg.Wait()
+		})
 	}
-	wg.Wait()
 }
 
 func Test_Destroy(t *testing.T) {
-	cacheDir := path.Join(os.Getenv("HOME"), "CacheHandlerTest/dir")
-	chTestArgs := initializeCacheHandlerTestArgs(t, &config.FileCacheConfig{EnableCRC: true}, cacheDir)
-	minObject1 := createObject(t, chTestArgs.bucket, "object_1", []byte("content of object_1"))
-	minObject2 := createObject(t, chTestArgs.bucket, "object_2", []byte("content of object_2"))
-	cacheHandle1, err := chTestArgs.cacheHandler.GetCacheHandle(minObject1, chTestArgs.bucket, true, 0)
-	require.NoError(t, err)
-	cacheHandle2, err := chTestArgs.cacheHandler.GetCacheHandle(minObject2, chTestArgs.bucket, true, 0)
-	require.NoError(t, err)
-	ctx := context.Background()
-	// Read to create and populate file in cache.
-	buf := make([]byte, 3)
-	_, cacheHit, err := cacheHandle1.Read(ctx, chTestArgs.bucket, minObject1, 4, buf)
-	require.NoError(t, err)
-	require.False(t, cacheHit)
-	_, cacheHit, err = cacheHandle2.Read(ctx, chTestArgs.bucket, minObject2, 4, buf)
-	require.NoError(t, err)
-	require.False(t, cacheHit)
-	err = cacheHandle1.Close()
-	require.NoError(t, err)
-	err = cacheHandle2.Close()
-	require.NoError(t, err)
-
-	err = chTestArgs.cacheHandler.Destroy()
-
-	assert.NoError(t, err)
-	// Verify the cacheDir is deleted.
-	_, err = os.Stat(path.Join(chTestArgs.cacheDir, util.FileCache))
-	assert.ErrorIs(t, err, os.ErrNotExist)
-	// Verify jobs statuses.
-	job1 := chTestArgs.jobManager.GetJob(minObject1.Name, chTestArgs.bucket.Name())
-	job2 := chTestArgs.jobManager.GetJob(minObject1.Name, chTestArgs.bucket.Name())
-	if job1 != nil {
-		assert.Contains(t, downloader.Completed, job1.GetStatus().Name)
+	tbl := []struct {
+		name                   string
+		fileCacheConfig        config.FileCacheConfig
+		cacheDir               string
+		expectedCacheHandleErr error
+		expectedJobStatus      []string
+	}{
+		{
+			name:                   "Non parallel downloads",
+			fileCacheConfig:        config.FileCacheConfig{EnableCRC: true},
+			cacheDir:               path.Join(os.Getenv("HOME"), "CacheHandlerTest/dir"),
+			expectedCacheHandleErr: nil,
+			expectedJobStatus:      []string{string(downloader.Completed)},
+		},
+		{
+			name: "Parallel downloads",
+			fileCacheConfig: config.FileCacheConfig{EnableCRC: true, EnableParallelDownloads: true,
+				ParallelDownloadsPerFile: 4, MaxParallelDownloads: 20, DownloadChunkSizeMB: 3},
+			cacheDir:               path.Join(os.Getenv("HOME"), "CacheHandlerTest/dir"),
+			expectedCacheHandleErr: fmt.Errorf("read via gcs"),
```

Comment:
Can you add a comment explaining why an error is returned for Parallel downloads?